pkg/cloud/amazon/aws_kms.go

```diff
@@ -128,38 +129,41 @@ func MakeAWSKMS(ctx context.Context, uri string, env cloud.KMSEnv) (cloud.KMS, e
 	}
 
 	region := kmsURIParams.region
-	awsConfig := &aws.Config{
-		Credentials: credentials.NewStaticCredentials(kmsURIParams.accessKey,
-			kmsURIParams.secret, kmsURIParams.tempToken),
+	if region == "" {
+		// TODO(adityamaru): Maybe use the KeyID to get the region, similar to how
+		// we infer the region from the bucket for s3_storage.
+		return nil, errors.New("could not find the aws kms region")
```

Comment:
is this going to be a user-visible behavior regression?

Reply:
Not a documented one. https://www.cockroachlabs.com/docs/stable/take-and-restore-encrypted-backups#aws-kms-uri-format. 

But i can do a little more digging once this merges.

Reply:
verified that this is not a regression. If the user does not specify a region, they hit this error (before this pr):
https://github.com/msbutler/cockroach/blob/butler-kms-no-region/pkg/cloud/amazon/aws_kms.go#L222

Reply:
Thanks for checking!